syntax = "proto3";

package book_suggestion;

message BookSuggestionRequest {
    Item item = 1;
}

message Item {
    string name = 1;
    int32 quantity = 2;
}

message BookSuggestionResponse {
    repeated Book books = 1;
}

message Book {
    string id = 1;
    string title = 2;
    string author = 3;
    string description = 4;
    int32 copies= 5;
    int32 copiesAvailable = 6;
    string category = 7;
    string img = 8;
    float price = 9;
}

service BookSuggestionService {
  rpc SuggestBook(BookSuggestionRequest) returns (BookSuggestionResponse);
}